fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java

```diff
@@ -287,6 +286,41 @@ public static boolean argumentTypeContainDecimalV3(String fnName, Type[] argumen
                 .anyMatch(t -> t.getItemType().isDecimalV3());
     }
 
+    // Multi parameters decimalV2 function will match decimalV3 function, and doesn't set decimalV3 precision&scale.
+    // like that, user input may: array_slice(decimalV2, tinyint), sr always match array_slice(decimalV3, bigint),
+    // because array_slice(decimalV2, tinyint) can't match array_slice(decimalV2, bigint) in IS_IDENTICAL mode, but
+    // only match array_slice(decimalV3, bigint) in IS_NONSTRICT_SUPERTYPE_OF mode (decimalV3 function is higher
+    // than decimalV2 function).
+    public static boolean argumentTypeContainDecimalV2(String fnName, Type[] argumentTypes) {
+        if (!FunctionSet.ARRAY_SLICE.equals(fnName)) {
+            return false;
+        }
+
+        if (Arrays.stream(argumentTypes).anyMatch(Type::isDecimalV2)) {
+            return true;
+        }
+
+        // check array child type
+        return Arrays.stream(argumentTypes).filter(Type::isArrayType).map(t -> (ArrayType) t)
```

Comment:
what if the nested array type?

Reply:
nest array type will use polymorphic function, it's right